pkg/kcp/provider/azure/vpcpeering/deleteVpcPeering.go

```diff
@@ -4,34 +4,41 @@ import (
 	"context"
 	"github.com/kyma-project/cloud-manager/pkg/composed"
 	azuremeta "github.com/kyma-project/cloud-manager/pkg/kcp/provider/azure/meta"
+	"github.com/kyma-project/cloud-manager/pkg/kcp/provider/azure/util"
+	"k8s.io/utils/ptr"
 )
 
 func deleteVpcPeering(ctx context.Context, st composed.State) (error, context.Context) {
 	state := st.(*State)
 	logger := composed.LoggerFromCtx(ctx)
 	obj := state.ObjAsVpcPeering()
 
-	if len(obj.Status.Id) == 0 {
-		logger.Info("VpcPeering deleted before Azure peering is created")
+	if state.localPeering == nil {
+		logger.Info("Skipping local peering deletion")
```

Comment:
If my domain understanding is correct, if we enter this block, the peering is already deleted. Under that assumption, this message is misleading.

Reply:
In that case I will leave old message "VpcPeering deleted before Azure peering is created"